modules/siddhi-extensions/regex/src/test/java/org/wso2/siddhi/extension/regex/GroupFunctionExtensionTestCase.java

```diff
@@ -0,0 +1,244 @@
+                "insert into outputStream;");
+        ExecutionPlanRuntime executionPlanRuntime = siddhiManager.createExecutionPlanRuntime
+                (inStreamDefinition + query);
+
+        executionPlanRuntime.start();
+        executionPlanRuntime.shutdown();
+    }
+
+    @Test (expected = ExecutionPlanValidationException.class)
+    public void testGroupFunctionExtension3() throws InterruptedException {
+        log.info("GroupFunctionExtensionTestCase TestCase with invalid datatype");
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String inStreamDefinition = "define stream inputStream (symbol int, price long, regex string, group int);";
+        String query = ("@info(name = 'query1') " +
+                "from inputStream " +
+                "select symbol , regex:group(regex, symbol, group) as aboutWSO2 " +
+                "insert into outputStream;");
+        ExecutionPlanRuntime executionPlanRuntime = siddhiManager.createExecutionPlanRuntime
+                (inStreamDefinition + query);
+
+        executionPlanRuntime.start();
+        executionPlanRuntime.shutdown();
+    }
+
+    @Test (expected = ExecutionPlanValidationException.class)
+    public void testGroupFunctionExtension4() throws InterruptedException {
+        log.info("GroupFunctionExtensionTestCase TestCase with invalid datatype");
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String inStreamDefinition = "define stream inputStream (symbol string, price long, regex string, "
+                + "group string);";
+        String query = ("@info(name = 'query1') " +
+                "from inputStream " +
+                "select symbol , regex:group(regex, symbol, group) as aboutWSO2 " +
+                "insert into outputStream;");
+        ExecutionPlanRuntime executionPlanRuntime = siddhiManager.createExecutionPlanRuntime
+                (inStreamDefinition + query);
+
+        executionPlanRuntime.start();
+        executionPlanRuntime.shutdown();
+    }
+
+    @Test
+    public void testGroupFunctionExtension5() throws InterruptedException {
+        log.info("GroupFunctionExtensionTestCase TestCase with null value");
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String inStreamDefinition = "define stream inputStream (symbol string, price long, regex string, group int);";
+        String query = ("@info(name = 'query1') " +
+                "from inputStream " +
+                "select symbol , regex:group(regex, symbol, group) as aboutWSO2 " +
+                "insert into outputStream;");
+        ExecutionPlanRuntime executionPlanRuntime = siddhiManager.createExecutionPlanRuntime
+                (inStreamDefinition + query);
+
+        InputHandler inputHandler = executionPlanRuntime.getInputHandler("inputStream");
+        executionPlanRuntime.start();
+        inputHandler.send(new Object[]{"21 products are produced within 10 years by WSO2 currently by WSO2 "
```

Comment:
Can't we do any sort of assertion here?

Reply:
yes, we don't have any sort of assertion here.

Reply:
why can't we assert? what are you trying to test here?

Reply:
When we input null value it will throw the ExecutionRuntimeException

Reply:
Ah i see ok